src/transformers/cache_utils.py

```diff
@@ -1673,6 +1683,7 @@ def __init__(
                 "config and it's not set to None."
             )
         self.max_cache_len = max_cache_len
+        self._sliding_window_max_len = min(config.sliding_window, max_cache_len)
```

Comment:
`HybridCache` had the right pattern, but some of the other hybrid caches did not: generation was crashing if we tried to generate a max length < sliding window length. Caught by one of the new tests.